AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/ads/max/nativead/ManualNativeLateBindingAdActivity.java

```diff
@@ -0,0 +1,120 @@
+import com.applovin.mediation.nativeAds.MaxNativeAdViewBinder;
+
+public class ManualNativeLateBindingAdActivity
+        extends BaseAdActivity
+{
+    private MaxNativeAdLoader nativeAdLoader;
+    private FrameLayout       nativeAdLayout;
+
+    private MaxAd nativeAd;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState)
+    {
+        super.onCreate( savedInstanceState );
+        setContentView( R.layout.activity_native_manual );
+        setTitle( R.string.activity_manual_native_late_binding_ad );
+
+        nativeAdLayout = findViewById( R.id.native_ad_layout );
+        setupCallbacksRecyclerView();
+
+        nativeAdLoader = new MaxNativeAdLoader( "YOUR_AD_UNIT_ID", this );
+        nativeAdLoader.setRevenueListener( ad -> {
+            logAnonymousCallback();
+
+            AdjustAdRevenue adjustAdRevenue = new AdjustAdRevenue( AdjustConfig.AD_REVENUE_APPLOVIN_MAX );
+            adjustAdRevenue.setRevenue( ad.getRevenue(), "USD" );
+            adjustAdRevenue.setAdRevenueNetwork( ad.getNetworkName() );
+            adjustAdRevenue.setAdRevenueUnit( ad.getAdUnitId() );
+            adjustAdRevenue.setAdRevenuePlacement( ad.getPlacement() );
+
+            Adjust.trackAdRevenue( adjustAdRevenue );
+        } );
+        nativeAdLoader.setNativeAdListener( new MaxNativeAdListener()
+        {
+            @Override
+            public void onNativeAdLoaded(final MaxNativeAdView nativeAdView, final MaxAd ad)
+            {
+                logAnonymousCallback();
+
+                // Cleanup any pre-existing native ad to prevent memory leaks.
+                if ( nativeAd != null )
+                {
+                    nativeAdLoader.destroy( nativeAd );
+                }
+
+                // Save ad for cleanup.
+                nativeAd = ad;
+
+                // Add ad view to view.
+                nativeAdLayout.removeAllViews();
+                nativeAdLayout.addView( nativeAdView );
+            }
+
+            @Override
+            public void onNativeAdLoadFailed(final String adUnitId, final MaxError error)
+            {
+                logAnonymousCallback();
+            }
+
+            @Override
+            public void onNativeAdClicked(final MaxAd ad)
+            {
+                logAnonymousCallback();
+            }
+        } );
+    }
+
+    @Override
+    protected void onDestroy()
+    {
+        // Must destroy native ad or else there will be memory leaks.
+        if ( nativeAd != null )
+        {
+            // Call destroy on the native ad from any native ad loader.
+            nativeAdLoader.destroy( nativeAd );
+        }
+
+        // Destroy the actual loader itself
+        nativeAdLoader.destroy();
+
+        super.onDestroy();
+    }
+
+    public void onShowAdClicked(View view)
+    {
+        nativeAdLoader.loadAd();
+        // Render the ad separately
+        nativeAdLoader.render( createNativeAdView(), nativeAd );
```

Comment:
I think it might be better to have two buttons, one `Load Ad` and another `Render Ad`. The `nativeAd` can be `null` here when `Show Ad` is first clicked. So, we should make sure `Render Ad` button is only enabled when an actual ad is loaded.

Reply:
Good idea